lib/pos-parser/nfa.ts

```diff
@@ -84,6 +84,31 @@ function arrayMatchCount(a : string[], b : string[]) : number {
     return count;
 }
 
+function posMatch(current : Token, next : Token|null, transitions : Record<string, Transition[]>) {
+    if (!current.pos)
+        return false;
+
+    // do not match part-of-speech for special token "$domain"
+    // "$value", on the other hand, is allowed (used in reverse_property templates)
+    if (current.token === '$domain')
+        return false;
+
+    if (!(current.pos in transitions))
+        return false;
+
+    if (current.pos === 'IN' && current.token === 'with')
+        return false;
+
+    if (current.pos.startsWith('V')) {
+        if (['has', 'have', 'had'].includes(current.token))
+            return next && ['VBN', 'VBG'].includes(next.pos!);
+        if (['is', 'are', 'was', 'were'].includes(current.token))
+            return false;
+    }
```

Comment:
I wonder if you should instead have multiple templates match the sentence, and use a scoring function to determine which wins...